RIOT-OS/RIOT
# cpu/stm32g4: add support + add nucleo-g474re board

### Contribution description

This PR adds support for the STM32 G4 family of CPUs and add support for the nucleo-g474re.

- What has been tested with success so far: i2c, spi, uart, rtc, rtt, timer, hwrng, flashpage
- ~What has been tested but doesn't fully work: flashpage. There's an issue with the raw mode.~
- What is configured but not tested in this port:
  - ~i2c: could be configured but doesn't work. Probably because of a I2C clock misconfiguration. I haven't looked in detail yet~
  -  ~spi~

The clock configuration uses a similar strategy as the one used in #14163. Depending on the one that is merged first, I'll factorize the 2.




### Testing procedure

- A green Murdock
- Run `./dist/tools/compile_and_test_for_board/compile_and_test_for_board.py . nucleo-g474re --jobs=4` and check the results are ok (will report later when done).




### Issues/PRs references

None
